src/java/picard/illumina/ClusterDataToSamConverter.java

```diff
@@ -101,10 +107,39 @@ public ClusterDataToSamConverter(final String runBarcode,
     }
 
     /**
+     *  Sets the SAM tag to use to store the molecular index bases.  If multiple molecular indexes exist, it will concatenate them
+     * and store them in this tag.
+     */
+    public ClusterDataToSamConverter withMolecularIndexTag(final String molecularIndexTag) {
```

Comment:
I like "with*" since it reads more naturally:

```
new ClusterDataToSamConverter(...).withMolecularIndexTag(...).withMolecularIndexQualityTag(...).withTagPerMolecularIndex(...)
```

versus

```
new ClusterDataToSamConverter(...).setMolecularIndexTag(...).setMolecularIndexQualityTag(...).setTagPerMolecularIndex(...)
```

Also, I don't see a lot of factories in Picard, three to be exact, so perhaps `with*` is reasonable?  As this is a style question, perhaps others @ktibbett or @tfenne could weigh in?